modules/crud/schema/schemaUpdate.go

```diff
@@ -0,0 +1,189 @@
+		return nil
+	}
+
+	schemaDb, ok := s.SchemaDoc[dbType]
+	if !ok {
+		return errors.New(dbType + " Db Not present in Schema")
+	}
+	schemaDoc, ok := schemaDb[col]
+	if !ok {
+		return errors.New(col + " Col Not present in Schema of " + dbType)
+	}
+
+	var temp interface{}
+	var err error
+	for key, doc := range updateDoc {
+		switch key {
+		case "set":
+			temp, err = s.validateSetOperation(doc, schemaDoc)
+			if err != nil {
+				return err
+			}
+		case "push":
+			temp, err = s.validateArrayOperations(doc, schemaDoc)
+			if err != nil {
+				return err
+			}
+		case "inc", "min", "max", "mul":
+			if err := validateMathOperations(doc, schemaDoc); err != nil {
+				return err
+			}
+		case "currentDate", "currentTimestamp":
+			temp, err = validateDateOperations(doc, schemaDoc)
+			if err != nil {
+				return err
+			}
+		default:
+			return errors.New(key + " Update operator not supported")
+		}
+		updateDoc[key] = temp
+	}
+	return nil
+}
+
+func (s *Schema) validateArrayOperations(doc interface{}, schemaDoc SchemaField) (interface{}, error) {
+
+	v, ok := doc.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("Schema math op wrong type passed expecting map[string]interface{}")
+	}
+
+	for fieldKey, fieldValue := range v {
+
+		schemaDocValue, ok := schemaDoc[fieldKey]
+		if !ok {
+			return nil, errors.New("field not found in schemaField")
+		}
+
+		switch t := fieldValue.(type) {
+		case []interface{}:
+			arr := make([]interface{}, len(t))
+			for index, value := range t {
+				newVal, err := s.checkType(value, schemaDocValue)
+				if err != nil {
+					return nil, err
+				}
+				arr[index] = newVal
+			}
+			return arr, nil
+		default:
+			return nil, errors.New("Schema Update Math Op. Wrong type ")
+		}
+	}
+
+	return nil, nil
+}
+
+func validateMathOperations(doc interface{}, schemaDoc SchemaField) error {
+
+	v, ok := doc.(map[string]interface{})
+	if !ok {
+		return errors.New("Schema math op wrong type passed expecting map[string]interface{}")
+	}
+
+	for fieldKey, fieldValue := range v {
+
+		schemaDocValue, ok := schemaDoc[fieldKey]
+		if !ok {
+			return errors.New("field not found in schemaField")
+		}
+
+		switch fieldValue.(type) {
+		case int:
+			switch schemaDocValue.Kind {
+			case TypeInteger:
+			default:
+				return errors.New("Integer wrong type wanted Integer")
+			}
+		case float32, float64:
+			switch schemaDocValue.Kind {
+			case TypeFloat:
+			default:
+				return errors.New("Float wrong type wanted Float")
+			}
+		default:
+			return errors.New("Schema Update Math Op. Wrong type ")
+		}
+	}
+
+	return nil
+}
+
+func validateDateOperations(doc interface{}, schemaDoc SchemaField) (interface{}, error) {
+
+	v, ok := doc.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("Schema math op wrong type passed expecting map[string]interface{}")
+	}
+	newMap := map[string]interface{}{}
+	var temp interface{}
+	for fieldKey, fieldValue := range v {
+
+		schemaDocValue, ok := schemaDoc[fieldKey]
```

Comment:
All you need to do is check if the kind is date time, thats it. No need to do anything else